src/shims/unix/socket.rs

```diff
@@ -1,15 +1,22 @@
+use std::cell::RefCell;
+use std::collections::VecDeque;
 use std::io;
+use std::rc::Rc;
 
 use crate::shims::unix::*;
 use crate::*;
 
 use self::fd::FileDescriptor;
 
+use rustc_middle::ty::TyCtxt;
+
 /// Pair of connected sockets.
-///
-/// We currently don't allow sending any data through this pair, so this can be just a dummy.
 #[derive(Debug)]
-struct SocketPair;
+struct SocketPair {
+    readbuf: Rc<RefCell<VecDeque<u8>>>,
+    writebuf: Rc<RefCell<VecDeque<u8>>>,
+    max_capacity: usize,
```

Comment:
Isn't max_capacity a property of the buffer, rather than the socket pair?

Currently it is anyway a constant so there's little point in carrying it here as a field, is there?

Reply:
Yup you are right, it should be a constant for buffer instead of a field.